client/web/src/enterprise/codeintel/uploads/hooks/queryLisfUploadFields.tsx

```
@@ -35,7 +35,9 @@ export const queryLisfUploadFields = (
         map(({ data }) => data),
         map(({ node }) => {
             if (!node || node.__typename !== 'LSIFUpload') {
-                throw new Error('No such LSIFUpload')
+                return new Error(
+                    'No such LSIFUpload. This can happen if the upload is associated with a nonexistent commit.'
```

Can we check to see if the commit is bad? I'm sure this could hide other error conditions, no?